### PR TITLE
test: bonjour e2e

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

yes

### Motivation / Use-Case

migrate to e2e option test

### Breaking Changes
no

### CODE REVIEW

File: test/server/bonjour-option.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    expect(mockPublish).toHaveBeenCalledTimes(1);
+    expect(mockPublish).toHaveBeenCalledWith({
+      name: `Webpack Dev Server ${os.hostname()}:${port}`,
+      port,
+      type: 'http',
+      subtypes: ['webpack'],
+    });
+    expect(mockUnpublishAll).toHaveBeenCalledTimes(0);
+  });
+});
```
Comment (alexander-akait) [inline test/server/bonjour-option.test.js:49]: Can we add `https` test too?

Comment (anshumanv) [reply test/server/bonjour-option.test.js:49]: yep on it

Comment (anshumanv) [reply test/server/bonjour-option.test.js:49]: we have hardcoded in `runBonjour` to use http, I think we should pick from config whether it's http or https

**Review by alexander-akait — DISMISSED** [review]

**Comment by anshumanv** [comment]
bonjour options will be implemented in a future pr, will add https tests with that

**Comment by anshumanv** [comment]
CI taking very long, running same task since 40 mins 😞

**Comment by alexander-akait** [comment]
It is freeze, something broken in tests

**Comment by anshumanv** [comment]
maybe server didn't close, lemme try skipping this

**Comment by anshumanv** [comment]
@alexander-akait green now 🟢

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**